Assets/Scripts/Player/PlayerController.cs

```diff
@@ -111,8 +118,63 @@ private void Update()
             Vector3 playerPosition = transform.position;
             _camera.transform.position = playerPosition + _cameraPosition;
             _camera.transform.LookAt(playerPosition);
+
+            // Check and makes character facing the target object when moving toward it
+            if (target != null) FaceTarget();
+        }
+
+        /// <summary>
+        /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+        /// (Won't conflict with character attacking enemy or moving on the map)
+        /// </summary>
+        #region Interactables Functions
+        
+        // This will make character moving to the target object
+        public void MovingToTarget(Interactable newTarget)
```

Comment:
I think this is something that can be combined with the regular movement logic, since it's basically the same

Reply:
Same as above, this should be called SetTarget instead of MovingToTarget. Sorry. This will make the character stop when it is close enough to the target by setting the stopping distance = the "radius" value of the object (can be tweaked)